### ISSUE TITLE
Add CI for all platforms

### ISSUE THREAD

**Issue by Youw** [libusb/libusb-cmake#7]
- [ ] macOS
- [ ] Windows MSVC/VisualStudio
- [ ] Windows MSVC/Ninja/nmake
- [ ] Windows Cygwin/MinGW
- [ ] (Fedora?) Cross-compile MinGW
- [ ] FreeBSD
- [ ] OpenBSD
- [ ] Android NDK

Any other platforms?

**Comment by mcuee** [comment]
I changed FreeBSD to NetBSD as FreeBSD has their own version of libusb under BSD licenses and it is part of the FreeBSD system (not an add-on).

**Comment by mcuee** [comment]
There are other platforms like Solaris and Haiku but I guess we may not want to deal with them.

Reference:
* https://github.com/libusb/libusb/issues/1099

**Comment by mcuee** [comment]
@Youw 

I have no ideas about WebAssembly + WebUSB support.
https://github.com/libusb/libusb/wiki#supported-environments

**Comment by mcuee** [comment]
@Youw 

For the CI action, it will be good to keep the binaries as well, at least for Windows (MSVC x86/x64/ARM/ARM64 and MinGW x86/x64).

**Comment by Youw** [comment]
> There are other platforms like Solaris and Haiku but I guess we may not want to deal with them.

My thoughs - right now it is not a part of CMake implementation. As soon as someone needs those (and adds its support to CMakeLists.txt) - we will think about a CI for it too.

**Comment by Youw** [comment]
> good to keep the binaries as well

I'm not so sure about that. Not any time soon.
Firstly - because CMake is not an official build system for libusb.
Secondly - the main purpose of this repo is to be able to _build_ libusb with CMake.

If someone needs a prebuilt binaries - that better be provided by a mainline libusb repo/be a part of its release process/etc.

---
Sure, it will be useful for some, e.g. temporary binaries saved as part of a PR (e.g. to verify a bugfix for someone or so).
But I don't want people to _expect_ to find prebuilt binaries here. And even the bugfix verification - that's more about PRs to libusb itself, to to libusb-cmake.

**Comment by mcuee** [comment]
@Youw 

I agree with you now, not to keep the binary as this is not official libusb. Rather we should improve libusb github action to keep Windows binaries, not in libusb-cmake.
